swim/handlers.go

```diff
@@ -121,9 +121,22 @@ func (n *Node) tickHandler(ctx json.Context, req *emptyArg) (*ping, error) {
 	return &ping{Checksum: n.memberlist.Checksum()}, nil
 }
 
-func (n *Node) adminJoinHandler(ctx json.Context, req *emptyArg) (*Status, error) {
-	n.memberlist.MakeAlive(n.address, util.TimeNowMS())
-	return &Status{Status: "rejoined"}, nil
+type Clock interface {
+	Now() time.Time
+}
+
+type globalClock struct{}
+
+func (c globalClock) Now() time.Time { return time.Now() }
+
+type handler func(json.Context, *emptyArg) (*Status, error)
+
+func (n *Node) adminJoinHandler(clock Clock) handler {
+	return func(ctx json.Context, req *emptyArg) (*Status, error) {
+		tInMS := clock.Now().UnixNano() / int64(time.Millisecond)
```

Comment:
I'd go so far to say that `type Clock` could expose a `IncarnationNumber()` method which hides the conversion of time to milliseconds.